Merge from upstream v14

Alchez opened the pull request:
**Changelogs:**
- https://github.com/frappe/frappe/releases/tag/v14.31.0
- https://github.com/frappe/frappe/releases/tag/v14.32.0
  - https://github.com/frappe/frappe/releases/tag/v14.32.1
- https://github.com/frappe/frappe/releases/tag/v14.33.0
  - https://github.com/frappe/frappe/releases/tag/v14.33.1

pstuhlmueller committed:
fix: Incorrect use of the Walrus operator

Incorrect use of the Walrus operator leads to unintended behavior for if-condition: "None" will be appended to cc.

(cherry picked from commit 2a80bb01ac1a41d5e5fcd70c7a37f62c5c93e8a9)

pstuhlmueller committed:
fix: sending mails to unintended recipients as cc

Security vulnerability: Unintentionally, all incoming emails are sent as CC to all users in a ToDo as "allocated_to" with the status "Open"

(cherry picked from commit 841557338b69d455a683ce4f02e8c76a12016b49)

deepeshgarg007 committed:
fix: Consider user perimission default in get_user_default function

(cherry picked from commit 32d3cc7277cbe6a5479f4bee893b41d9634a6470)

shariquerik committed:
fix: cannot restore cancelled document if workflow is active

(cherry picked from commit d2f529462bb2425a51205b2e16c521d8c44734c6)

ruchamahabal committed:
fix: don't hide tab with dashboard if there is a visible section/control

(cherry picked from commit 08c02907771f06258ae25e59897bd7de59a1e922)

nextchamp-saqib committed:
feat: redis cache decorator (#20452)

* feat: redis cache decorator

* fix: review changes

* fix: remove unintentional changes

* fix: remove unintentional changes

* refactor: cleanup and simplify code for redis

AIs suck

* fix: bug

* test: redis cache

* fix: remove unused import

* feat: make redis cache user specific

redis cache utils already support this, extending so everyone can use it

* feat: support @redis_cache without params

* test: flake in request site cache test

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit 32dbbb47bfc5fa89647befd1b3cdf21f5c27b9f9)

AHasanin committed:
fix(grid row): fix update_docfield_property function not updating grid row

(cherry picked from commit ed5ce1241ffbc99c55c69f2504e9b9ca89a927ff)

AHasanin committed:
fix(grid row): fix prettier hook check

(cherry picked from commit 60a5dc82c3b7b3637addd0c803ea51b38cb1039c)

mergify[bot] committed:
chore: add insights banner (#20487) (#20489)

* chore: add insights banner

* chore: cleanup message and title

[skip ci]

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit e756d417901b755ff2e65de97dddca06d25ee658)

Co-authored-by: Saqib Ansari <nextchamp.saqib@gmail.com>

mergify[bot] committed:
feat: Disable Sharing globally (backport #20318) (#20492)

* feat: Disable Sharing globally (#20318)

* feat: Disable Sharing globally

- Checkbox in System Settings
- If disabled, avoid share UI render
- Share APIs return None (non-obstructing) if share APIs are invoked

* feat: Settings checkbox must toggle share permission globally

- Treat feature like a perm toggler. Essentially noone is allowed to explicity share anything
- Implicit sharing via `ignore_share_permissions` is allowed. Devs can decide where sharing should happen under the hood
- UI is made read only and not hidden. Users must see who doc is already shared with
- Make sure perm APIs used by share feature return false if sharing is disabled
- Rename checkbox to `Disable Document Sharing`

* test: (server side) Impact of disabling sharing on APIs

- Also, fix missed system setting rename in `assign_to`

* fix: Inform assigner if assignee lacks perms and sharing is disabled

- misc: readable conditions

* fix: throw instead of msgprint

* fix: Typo and appropriate message for `throw`

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit 90f8f945b4f830d458ad453f81fc95fa3dfe9f75)

* test: fix test case to modified behaviour

We throw instead of showing warning now

---------

Co-authored-by: Marica <maricadsouza221197@gmail.com>
Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: date field shouldn't be formatted for TZ (#20486) (#20490)

Date fields aren't timezone aware.

(cherry picked from commit 63338a3806281a2933fac7fb158b730c1c64bca9)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: nestedset rename (#20498) (#20499)

(cherry picked from commit 45c86e2ff843f4dfdd35666c224f3b19845fbd8b)

Co-authored-by: Ankush Menat <ankush@frappe.io>

barredterra committed:
feat: make workflow state translatable (#20412)

Backport of https://github.com/frappe/frappe/pull/20326

sagarvora committed:
fix: escape HTML instead of sanitizing

(cherry picked from commit a56ea73b7d048c1536bc9957ba6e9cde2d1ddc3f)

sagarvora committed:
Merge pull request #20512 from frappe/mergify/bp/version-14-hotfix/pr-20508

fix: escape HTML instead of sanitizing (backport #20508)

barredterra committed:
fix: bulk update using doc method, check perms (#20522)

* fix: bulk update

* fix: always check permission

(cherry picked from commit 9758781f809a60fb1f1c3c29ad746f98478d6a3d)

sagarvora committed:
Merge pull request #20524 from frappe/mergify/bp/version-14-hotfix/pr-20522

fix: bulk update (backport #20522)

surajshetty3416 committed:
fix: Check if reference_name is set

(cherry picked from commit aab37e0a6ca59727edf7c48ee9bc64d09677fcea)

surajshetty3416 committed:
Merge pull request #20531 from frappe/mergify/bp/version-14-hotfix/pr-20115

fix(email): unintended emails sent as cc for todo allocated_to users (backport #20115)

ankush committed:
fix: Handle JsBarcode exceptions (#20532)

JsBarcode exceptions prevent entire page from loading. Instead of that
catch error and show it in helpbox so user can correct the barcode if
required.

Steps to reproduce:
1. Add barcode field
2. Set barcode type in options
3. add invalid barcode and save

mergify[bot] committed:
perf: Faster address query with explicit joins (backport #20537) (#20540)

* perf: Faster address query with explicit joins

This querry is particularly problamatic when there are OR conditions due
to shared docs.

```sql
ANALYZE SELECT `tabAddress`.name,
       `tabAddress`.city,
       `tabAddress`.country
FROM `tabAddress`
join `tabDynamic Link` on (`tabDynamic Link`.parent = `tabAddress`.name AND `tabDynamic Link`.parenttype = 'Address')
WHERE
  `tabDynamic Link`.link_doctype = 'Customer'
  AND `tabDynamic Link`.link_name = 'CUSTOMER NAME'
  AND coalesce(`tabAddress`.disabled, 0) = 0
  AND (`tabAddress`.`country` like '%%'
       OR `tabAddress`.`state` like '%%'
       OR `tabAddress`.`name` like '%%'
       OR `tabAddress`.`name` like '%%')
  AND ((((coalesce(`tabAddress`.`branch`, '')=''
          OR `tabAddress`.`branch` in ('something')))))
  OR (`tabAddress`.name in ('SOME SHARED DOC'))
ORDER BY if(locate('', `tabAddress`.name), locate('', `tabAddress`.name), 99999),
         `tabAddress`.idx DESC,
         `tabAddress`.name
LIMIT 0, 20;
```

**Before:** Never terminates and reads entire table with millions of lines :LOL:

**After:** Reads ~100 rows max.

```
*************************** 1. row ***************************
           id: 1
  select_type: SIMPLE
        table: tabDynamic Link
         type: index_merge
possible_keys: parent,link_doctype_link_name_index,link_name
          key: link_name,link_doctype_link_name_index,parent
      key_len: 563,1126,563
          ref: NULL
         rows: 40
       r_rows: 79.00
     filtered: 100.00
   r_filtered: 100.00
        Extra: Using union(intersect(link_name,link_doctype_link_name_index),parent); Using where; Using temporary; Using filesort
*************************** 2. row ***************************
           id: 1
  select_type: SIMPLE
        table: tabAddress
         type: eq_ref
possible_keys: PRIMARY,selco_branch
          key: PRIMARY
      key_len: 562
          ref: _900a733bdc3bb9ed.tabDynamic Link.parent
         rows: 1
       r_rows: 1.00
     filtered: 100.00
   r_filtered: 100.00
        Extra: Using where
2 rows in set (0.001 sec)
```

<!--

Some key notes before you open a PR:

 1. Select which branch should this PR be merged in?
 2. PR name follows [convention](http://karma-runner.github.io/4.0/dev/git-commit-msg.html)
 3. All tests pass locally, UI and Unit tests
 4. All business logic and validations must be on the server-side
 5. Update necessary Documentation
 6. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes

Also, if you're new here

- Documentation Guidelines => https://github.com/frappe/erpnext/wiki/Updating-Documentation

- Contribution Guide => https://github.com/frappe/frappe/blob/develop/.github/CONTRIBUTING.md

- Pull Request Checklist => https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

-->

> Please provide enough information so that others can review your pull request:

<!-- You can skip this if you're fixing a typo or updating existing documentation -->

> Explain the **details** for making this change. What existing problem does the pull request solve?

<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

> Screenshots/GIFs

<!-- Add images/recordings to better visualize the change: expected/current behviour -->

(cherry picked from commit 09e32c3085559848e73c15e4db14163f8e447a88)

* test: address query

(cherry picked from commit fba3497ba0f2c549453d6cae3c6cc553494a7a22)

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

Jainysh committed:
fix: change z-index of freeze component to make it appear above all components

changed the z-index value of #freeze id to 9999 so that it appears above all UI components

fixes #20538

(cherry picked from commit 96395d60dbc405b1fda1be3fe37ec7968aa8c1b9)

shariquerik committed:
fix(UI): align link cards & charts on workspace

(cherry picked from commit b72ec114eeed6de1bc77cedc9b5cc540ae3dd769)

ankush committed:
fix: fix address query for postgres

refer https://github.com/frappe/frappe/pull/20537#ref-pullrequest-1645575433

(cherry picked from commit 10dcbc5ecf2aa22b128a71a52061eb3a1c3da5f7)

shariquerik committed:
Merge branch 'version-14-hotfix' into mergify/bp/version-14-hotfix/pr-20549

shariquerik committed:
Merge pull request #20550 from frappe/mergify/bp/version-14-hotfix/pr-20549

fix(UI): align link cards & charts on workspace (backport #20549)

shariquerik committed:
fix: get workflow_state_fieldname instead of setting workflow_state to none

(cherry picked from commit ee9cb43835ee4f4cd14a12784e404a682500e6c7)

mergify[bot] committed:
chore: remove excessive whitespace (#20544) (#20555)

to make the linter happy

[skip ci]

(cherry picked from commit ae3b3ebb174d4709f01f002014ebaa0930632aab)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
test: Kanban Test fails to remove System Manager role (#20505) (#20551)

* fix: Kanban Test fails to remove System Manager role

- As there's only one user with System Manager role, role removal is reverted
- Add another user with this role, so that role removal on test user works

* chore: Reuse `create_test_user` util

* fix: user switching in kanban test

---------

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>
(cherry picked from commit f206b1582e31108efdd90893270b57ab259aace5)

Co-authored-by: Marica <maricadsouza221197@gmail.com>

mergify[bot] committed:
feat(util): `get_table_name`: wrap in backticks (#20553) (#20556)

(cherry picked from commit a1aaed0a5f23b8a52e0f075ca953e4102679102e)

Co-authored-by: Sabu Siyad <hello@ssiyad.com>

ruchamahabal committed:
fix: hide chart and heatmap on dashboard reset

(cherry picked from commit b443bca2d58c793fe4e5061fded9ff8b31aee10a)

mergify[bot] committed:
fix: rewrite query for postgres (#20557) (#20559)

(cherry picked from commit fa32b610d661bae0ca935cf3a4f7c3f5c3ecac66)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ruchamahabal committed:
fix: heatmap not reset on dashboard refresh

(cherry picked from commit e6b09ec771e6fd1146bd95ad4e9fb42ce19c0927)

mergify[bot] committed:
fix: removed unnecessary usage of `@frappe.whitelist` (#20503) (#20563)

(cherry picked from commit d2bde19b5c967b3980ce5175eb74e70e9e74eac4)

Co-authored-by: Daizy Modi <modidaizy5217@gmail.com>

mergify[bot] committed:
fix: allow `reset_otp_secret` only if Two Factor Auth is enabled (#20506) (#20561)

* fix: display `Reset OTP Secret` button only if Two factor Auth is enabled

* fix: added validations and fetched value from cached doc

* fix: linter changes

(cherry picked from commit 06580bdbff9a8f86709e52c82afe0cb9da2dc1d4)

Co-authored-by: Daizy Modi <modidaizy5217@gmail.com>

ruchamahabal committed:
fix: use chart type passed to `render_graph` on form dashboards

(cherry picked from commit d586d283c56eb971e548ef0326375c8c0b398d96)

mergify[bot] committed:
fix: allowed only POST and PUT methods in `rename_doc` (#20504) (#20565)

(cherry picked from commit 3db1c1aea00cbc79559a73c5cc9638f99ea07cc8)

Co-authored-by: Daizy Modi <modidaizy5217@gmail.com>

shariquerik committed:
Merge pull request #20567 from frappe/mergify/bp/version-14-hotfix/pr-20460

fix: cannot restore cancelled document if workflow is active (backport #20460)

mergify[bot] committed:
feat: list-view click and drag on check box to select multiple rows (backport #20457) (#20568)

Co-authored-by: Ponnusamy <95607086+Ponnusamy1-V@users.noreply.github.com>

mergify[bot] committed:
fix: content_type can be `None` during file upload (#20572) (#20574)

closes https://github.com/frappe/frappe/issues/20571

(cherry picked from commit b66c3d9106faa7db94960a9c2e45b380219143ad)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: unsubscribe from list_update before resubbing (#20450) (#20581)

resubbing can result in multiple events being fired, so unsubscribe all
of them before re-subscribing.

missed in https://github.com/frappe/frappe/pull/20423

(cherry picked from commit 6b3d283cf7ac7ce26907c67fe229fecf356388ef)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
perf: unsubscribe from list_update events (#20423) (#20580)

(cherry picked from commit 516540ede9bee9b0c5e61c3984376e9daeb190ce)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
build: bump pymysql (#20478) (#20576)

Actual fix for this bandaid fix: https://github.com/frappe/frappe/pull/20475

Keeping pymysql hard pinned until we have better way to get last full
query.

(cherry picked from commit 024faff02505cc5764d822ecab718814ea75f242)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: Drop message_id index before migrating email queue (#20376) (#20578)

[skip c]

(cherry picked from commit 7ed43e2a68082ffa36933a0abf97fc7002696944)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: email linking and message_id indexing (#20356) (#20579)

* fix: find communication regardless of system reply

* perf: convert and index message_id

(cherry picked from commit fdcdb61a2dc5ae8306fc67e406da8d1c6b9cbd80)

Co-authored-by: Ankush Menat <ankush@frappe.io>

shariquerik committed:
Merge pull request #20582 from frappe/mergify/bp/version-14-hotfix/pr-20461

fix: don't hide tab with dashboard if there is a visible section/control (backport #20461)

ankush committed:
Merge pull request #20586 from frappe/version-14-hotfix

chore: release v14

frappe-bot committed:
chore(release): Bumped to Version 14.31.0

# [14.31.0](https://github.com/frappe/frappe/compare/v14.30.0...v14.31.0) (2023-04-04)

### Bug Fixes

* allow `reset_otp_secret` only if Two Factor Auth is enabled ([#20506](https://github.com/frappe/frappe/issues/20506)) ([#20561](https://github.com/frappe/frappe/issues/20561)) ([ca486c4](https://github.com/frappe/frappe/commit/ca486c4636f8236bf0e711d65abe0324b467fbed))
* allowed only POST and PUT methods in `rename_doc` ([#20504](https://github.com/frappe/frappe/issues/20504)) ([#20565](https://github.com/frappe/frappe/issues/20565)) ([fd1f6fe](https://github.com/frappe/frappe/commit/fd1f6fe392acdf8b0206852d2ddb9e579cd822e2))
* bulk update using doc method, check perms ([#20522](https://github.com/frappe/frappe/issues/20522)) ([2237968](https://github.com/frappe/frappe/commit/2237968527668ec956e3c81d62d9867d5a047b0a))
* cannot restore cancelled document if workflow is active ([547efe3](https://github.com/frappe/frappe/commit/547efe3bbe889f9ddcb67f06e44ec5d1894fd85b))
* Check if reference_name is set ([2e9068b](https://github.com/frappe/frappe/commit/2e9068bf418b1434a4c4aedc7f77589b3c695a58))
* content_type can be `None` during file upload ([#20572](https://github.com/frappe/frappe/issues/20572)) ([#20574](https://github.com/frappe/frappe/issues/20574)) ([ebc6059](https://github.com/frappe/frappe/commit/ebc6059bd25e324426f0f936d242aba11499d6be))
* date field shouldn't be formatted for TZ ([#20486](https://github.com/frappe/frappe/issues/20486)) ([#20490](https://github.com/frappe/frappe/issues/20490)) ([07b7b46](https://github.com/frappe/frappe/commit/07b7b46ac1b61790c5758917e0d9976382ca0aef))
* don't hide tab with dashboard if there is a visible section/control ([084e8af](https://github.com/frappe/frappe/commit/084e8af34f7d7564d2e966bda66a7b4ae5581102))
* Drop message_id index before migrating email queue ([#20376](https://github.com/frappe/frappe/issues/20376)) ([#20578](https://github.com/frappe/frappe/issues/20578)) ([394c232](https://github.com/frappe/frappe/commit/394c232df94200ccfb4dd1e9a0501555708774d9))
* email linking and message_id indexing ([#20356](https://github.com/frappe/frappe/issues/20356)) ([#20579](https://github.com/frappe/frappe/issues/20579)) ([5ae94ef](https://github.com/frappe/frappe/commit/5ae94efaf89312b4b11a28a4b32f1431e8a53e19))
* escape HTML instead of sanitizing ([2269d6e](https://github.com/frappe/frappe/commit/2269d6e656014ff17e88b9747d529478d8163ecc))
* fix address query for postgres ([f01566a](https://github.com/frappe/frappe/commit/f01566aab2c26bfe7d1b2e2f44aff7a2d97f05fb)), closes [/github.com/frappe/frappe/pull/20537#ref-pullrequest-1645575433](https://github.com//github.com/frappe/frappe/pull/20537/issues/ref-pullrequest-1645575433)
* get workflow_state_fieldname instead of setting workflow_state to none ([bac4e9b](https://github.com/frappe/frappe/commit/bac4e9b7061fd77ecb3038b2b557b132056ad91c))
* Handle JsBarcode exceptions ([#20532](https://github.com/frappe/frappe/issues/20532)) ([b1c3d8b](https://github.com/frappe/frappe/commit/b1c3d8b5d91bb0c19842df8a3dcf48166be661c4))
* heatmap not reset on dashboard refresh ([3c0659b](https://github.com/frappe/frappe/commit/3c0659b3a706a2363549fa0472ef21149c317d34))
* hide chart and heatmap on dashboard reset ([cec7a73](https://github.com/frappe/frappe/commit/cec7a7330d28a636792b0a67d2a593a0b95e45bc))
* Incorrect use of the Walrus operator ([a70a5ca](https://github.com/frappe/frappe/commit/a70a5cad141dd22638057fbc2b1a7f58e36e0407))
* nestedset rename ([#20498](https://github.com/frappe/frappe/issues/20498)) ([#20499](https://github.com/frappe/frappe/issues/20499)) ([b831392](https://github.com/frappe/frappe/commit/b831392a3a3d4a9e6c76ea42a42e791bcee3dbdb))
* removed unnecessary usage of `[@frappe](https://github.com/frappe).whitelist` ([#20503](https://github.com/frappe/frappe/issues/20503)) ([#20563](https://github.com/frappe/frappe/issues/20563)) ([0af8315](https://github.com/frappe/frappe/commit/0af8315c4a8d50b2f508be80b98e454aacc4f0ac))
* rewrite query for postgres ([#20557](https://github.com/frappe/frappe/issues/20557)) ([#20559](https://github.com/frappe/frappe/issues/20559)) ([d754be5](https://github.com/frappe/frappe/commit/d754be50853141bc65225e5cb8c0b324c4dec8fe))
* sending mails to unintended recipients as cc ([1c47643](https://github.com/frappe/frappe/commit/1c47643a6cad6dfcb2b1cec66a517639e0113bd5))
* **UI:** align link cards & charts on workspace ([82cfd33](https://github.com/frappe/frappe/commit/82cfd3362f9beb5ad29c1d28b46be244bc45f8ef))
* unsubscribe from list_update before resubbing ([#20450](https://github.com/frappe/frappe/issues/20450)) ([#20581](https://github.com/frappe/frappe/issues/20581)) ([7866605](https://github.com/frappe/frappe/commit/7866605b8bcd5b3af1a089a64b344757f872e11f))
* use chart type passed to `render_graph` on form dashboards ([015f7db](https://github.com/frappe/frappe/commit/015f7dbadffbaf86f16f40963183f0dec7397006))

### Features

* Disable Sharing globally (backport [#20318](https://github.com/frappe/frappe/issues/20318)) ([#20492](https://github.com/frappe/frappe/issues/20492)) ([67a537c](https://github.com/frappe/frappe/commit/67a537cf1257f1355cb5d5a2d4b134b85cd0bb9d))
* list-view click and drag on check box to select multiple rows (backport [#20457](https://github.com/frappe/frappe/issues/20457)) ([#20568](https://github.com/frappe/frappe/issues/20568)) ([f757a37](https://github.com/frappe/frappe/commit/f757a375d6b6394fd7c760f7327be2303ffd0697))
* make workflow state translatable ([#20412](https://github.com/frappe/frappe/issues/20412)) ([8675789](https://github.com/frappe/frappe/commit/86757895f568b2b9009c904c42d2d5597a8ae1ad))
* **util:** `get_table_name`: wrap in backticks ([#20553](https://github.com/frappe/frappe/issues/20553)) ([#20556](https://github.com/frappe/frappe/issues/20556)) ([5c56dff](https://github.com/frappe/frappe/commit/5c56dff68629a570d6fe930c74ac7c7125acce5f))

### Performance Improvements

* Faster address query with explicit joins (backport [#20537](https://github.com/frappe/frappe/issues/20537)) ([#20540](https://github.com/frappe/frappe/issues/20540)) ([2ef7ef5](https://github.com/frappe/frappe/commit/2ef7ef50519e578ce4e31c6e077439a20294deed))
* unsubscribe from list_update events ([#20423](https://github.com/frappe/frappe/issues/20423)) ([#20580](https://github.com/frappe/frappe/issues/20580)) ([8a63144](https://github.com/frappe/frappe/commit/8a63144d6d6f3330913458533315bb3cbda12d1d))

deepeshgarg007 committed:
chore: Update server side method

(cherry picked from commit 0a55ac8f0774baac1fb06fbd60e5529ecf9561e3)

mergify[bot] committed:
perf: Don't re-initate sessions in realtime.py (#20577)

Session is already created for all requests in app.py -> init_request -> HTTPRequest()

(cherry picked from commit 3f717f2fbdc13571d409a99f94df25c78aa5886b)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: dont create communication for auto repeat notification (#20606) (#20611)

* fix: dont create communication upon auto repeat notification

creating communication for notification doesnt make sense

* test: rename test_notification_is_attached -> test_email_notification

now tests for email queue creation

(cherry picked from commit a3bc93e92899a22f53d3919376026661879e6877)

Co-authored-by: Ritwik Puri <ritwikpuri5678@gmail.com>

deepeshgarg007 committed:
fix: Consider global default in user perm

(cherry picked from commit e805497681382f0b8734fd6cfd56cd5856f64888)

deepeshgarg007 committed:
test: Add test for user perm defaults

(cherry picked from commit 087caff4cdb889c61ed32ac83dc7b9756ef77c5f)

# Conflicts:
#	frappe/tests/test_defaults.py

deepeshgarg007 committed:
test: Create user

(cherry picked from commit 5527048592048d9e1dd5c9908bf9a29332451735)

NagariaHussain committed:
fix: minor oauth doctypes enhancements (#20613)

* fix: remove extra`/` from description

* fix(ux): show bearer token status in list view

(cherry picked from commit a48b0f3cf8559d056804ae9500ea12fd19147dd0)

# Conflicts:
#	frappe/integrations/doctype/oauth_client/oauth_client.json

mergify[bot] committed:
fix: german translation of workflow state (#20609) (#20616)

(cherry picked from commit 3a76a771a9a19bed7b87c33729e38a92bb97cc71)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

mergify[bot] committed:
feat: make report name translatable (#20608) (#20618)

(cherry picked from commit cd670bf78fc49d8c0c48319516f456ee08ff3c8f)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

barredterra committed:
fix: improved validation in `add_comment` (#20520)

(cherry picked from commit e5b1b8d681240cf8d67bfeb25e2ed508afe30cec)

deepeshgarg007 committed:
test: Set defualts

(cherry picked from commit 68b1051f69430d934f08fe7725370ed504d0daeb)

deepeshgarg007 committed:
test: Update test

(cherry picked from commit 5d61ed2d8f268345cf0fc457c9124778befef4d5)

# Conflicts:
#	frappe/tests/test_defaults.py

mergify[bot] committed:
fix: also build jsx files (#20624) (#20628)

(cherry picked from commit 55867282adf798a4089062905fe29c069799d63b)

Co-authored-by: Mohammad Hussain Nagaria <34810212+NagariaHussain@users.noreply.github.com>

shariquerik committed:
fix: child row form should be above freeze screen

sidebar overlay was also getting affected

(cherry picked from commit 75ba9b40d8a0e18cf267d8b42cafc6053a85b574)

shariquerik committed:
Merge pull request #20630 from frappe/mergify/bp/version-14-hotfix/pr-20541

fix: change z-index of freeze component (backport #20541)

shariquerik committed:
Merge pull request #20626 from frappe/mergify/bp/version-14-hotfix/pr-20520

fix: improved validation in `add_comment` (backport #20520)

shariquerik committed:
chore: resolved conflicts

shariquerik committed:
Merge branch 'version-14-hotfix' into mergify/bp/version-14-hotfix/pr-20613

pstuhlmueller committed:
fix(ListView): Evaluate sort_field sort_order within listviews based on DocTypes Definition (#20482)

(cherry picked from commit 796250577987579bb1f2eb7699be9baabe0bad18)

deepeshgarg007 committed:
chore: Only run on mariadb

(cherry picked from commit c0ae00168dfac16f8b12cd2ea7985fc4281d65fb)

shariquerik committed:
Merge pull request #20614 from frappe/mergify/bp/version-14-hotfix/pr-20613

fix: minor oauth doctypes enhancements (backport #20613)

shariquerik committed:
Merge pull request #20633 from frappe/mergify/bp/version-14-hotfix/pr-20482

fix(ListView): Evaluate sort_field sort_order within listviews based on DocTypes Definition (backport #20482)

shariquerik committed:
Merge pull request #20635 from frappe/mergify/bp/version-14-hotfix/pr-20470

fix(grid row): fix update_docfield_property function not updating grid row (backport #20470)

mergify[bot] committed:
fix: log requests even if no response (#20638) (#20640)

(cherry picked from commit a3a9e40aa4f78a1c66457a9c25f4194f42c003f2)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix(pretty-date): plural form when the value is 1 (#20619) (#20642)

(cherry picked from commit 6248d8f06283ed6079c33dfe115c19aa93c0d7c9)

Co-authored-by: Bread Genie <63963181+BreadGenie@users.noreply.github.com>

mergify[bot] committed:
fix: validate if doctype exists before syncing customisations (backport #20598) (#20645)

* fix: validate if doctype exists before syncing customisations (#20598)

* fix: validate if doctype exists before syncing customisations

* refactor: rename and add default

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit 26e73208d596ebc8c1f6b8cad82528ba616778c8)

# Conflicts:
#	frappe/modules/utils.py

* chore: conflicts

---------

Co-authored-by: Saurabh <saurabh6790@gmail.com>
Co-authored-by: Ankush Menat <ankush@frappe.io>

deepeshgarg007 committed:
chore: Resolve conflicts

mergify[bot] committed:
fix: better permission error for query_report (#20643) (#20647)

(cherry picked from commit 08732e50dbbe881b20254667b97d67bc5b7b8d15)

Co-authored-by: Smit Vora <smitvora203@gmail.com>

deepeshgarg007 committed:
Merge pull request #20636 from frappe/mergify/bp/version-14-hotfix/pr-20375

fix: Consider user permission default in get_user_default function (backport #20375)

mergify[bot] committed:
fix: track workspace changes (#20649) (#20650)

(cherry picked from commit 75b32f90abe71924a1e63ec87fd4f5c97eab5bca)

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix!: contact us email reply (backport #20542) (#20584)

* fix: contact us email reply

(cherry picked from commit 67de2a34ac94b9fa6199c2525ac97b66549e47f0)

* fix: remove methods arg from tate_limit decorator

* feat(minor): send a copy of message to the sender/contactee
* feat: validate sender email address for contact us page

(cherry picked from commit 1cb5bfe8755da0a653c6846e407e0374c19ba1f9)

* refactor: remove frappe.send_message js util

(cherry picked from commit 0e4909b28b31c6bec85c48cda3972a973f3b7cb4)

* fix: suppress outgoing email error for contact-us page

(cherry picked from commit cbe0ce37d3f344012db9f7502699ccd1786ad53d)

* Revert "refactor: remove frappe.send_message js util"

This reverts commit 852cd528d6c46c69ed384502fdf4db934a439982.

---------

Co-authored-by: phot0n <ritwikpuri5678@gmail.com>

Commit:
Merge remote-tracking branch 'frappe/version-14' into merge-v14

ghost commented:
### Chore

* add insights banner (#20487) (#20489) ([f459fb0](https://github.com/ParsimonyGit/frappe/commit/f459fb0e1b39b133b029b5d63114a6232e33b876))
* remove excessive whitespace (#20544) (#20555) ([c8beea2](https://github.com/ParsimonyGit/frappe/commit/c8beea244c88da194bc358565a49b328137cea09))


### Features

* Disable Sharing globally (backport #20318) (#20492) ([67a537c](https://github.com/ParsimonyGit/frappe/commit/67a537cf1257f1355cb5d5a2d4b134b85cd0bb9d))
* make workflow state translatable (#20412) ([8675789](https://github.com/ParsimonyGit/frappe/commit/86757895f568b2b9009c904c42d2d5597a8ae1ad))
*  **util:** `get_table_name`: wrap in backticks (#20553) (#20556) ([5c56dff](https://github.com/ParsimonyGit/frappe/commit/5c56dff68629a570d6fe930c74ac7c7125acce5f))
* list-view click and drag on check box to select multiple rows (backport #20457) (#20568) ([f757a37](https://github.com/ParsimonyGit/frappe/commit/f757a375d6b6394fd7c760f7327be2303ffd0697))


### Bug Fixes

* date field shouldn't be formatted for TZ (#20486) (#20490) ([07b7b46](https://github.com/ParsimonyGit/frappe/commit/07b7b46ac1b61790c5758917e0d9976382ca0aef))
* nestedset rename (#20498) (#20499) ([b831392](https://github.com/ParsimonyGit/frappe/commit/b831392a3a3d4a9e6c76ea42a42e791bcee3dbdb))
* escape HTML instead of sanitizing ([2269d6e](https://github.com/ParsimonyGit/frappe/commit/2269d6e656014ff17e88b9747d529478d8163ecc))
* bulk update using doc method, check perms (#20522) ([2237968](https://github.com/ParsimonyGit/frappe/commit/2237968527668ec956e3c81d62d9867d5a047b0a))
* Incorrect use of the Walrus operator ([a70a5ca](https://github.com/ParsimonyGit/frappe/commit/a70a5cad141dd22638057fbc2b1a7f58e36e0407))
* sending mails to unintended recipients as cc ([1c47643](https://github.com/ParsimonyGit/frappe/commit/1c47643a6cad6dfcb2b1cec66a517639e0113bd5))
* Check if reference_name is set ([2e9068b](https://github.com/ParsimonyGit/frappe/commit/2e9068bf418b1434a4c4aedc7f77589b3c695a58))
* Handle JsBarcode exceptions (#20532) ([b1c3d8b](https://github.com/ParsimonyGit/frappe/commit/b1c3d8b5d91bb0c19842df8a3dcf48166be661c4))
* fix address query for postgres ([f01566a](https://github.com/ParsimonyGit/frappe/commit/f01566aab2c26bfe7d1b2e2f44aff7a2d97f05fb))
*  **UI:** align link cards & charts on workspace ([82cfd33](https://github.com/ParsimonyGit/frappe/commit/82cfd3362f9beb5ad29c1d28b46be244bc45f8ef))
* rewrite query for postgres (#20557) (#20559) ([d754be5](https://github.com/ParsimonyGit/frappe/commit/d754be50853141bc65225e5cb8c0b324c4dec8fe))
* removed unnecessary usage of `@frappe.whitelist` (#20503) (#20563) ([0af8315](https://github.com/ParsimonyGit/frappe/commit/0af8315c4a8d50b2f508be80b98e454aacc4f0ac))
* allow `reset_otp_secret` only if Two Factor Auth is enabled (#20506) (#20561) ([ca486c4](https://github.com/ParsimonyGit/frappe/commit/ca486c4636f8236bf0e711d65abe0324b467fbed))
* allowed only POST and PUT methods in `rename_doc` (#20504) (#20565) ([fd1f6fe](https://github.com/ParsimonyGit/frappe/commit/fd1f6fe392acdf8b0206852d2ddb9e579cd822e2))
* cannot restore cancelled document if workflow is active ([547efe3](https://github.com/ParsimonyGit/frappe/commit/547efe3bbe889f9ddcb67f06e44ec5d1894fd85b))
* get workflow_state_fieldname instead of setting workflow_state to none ([bac4e9b](https://github.com/ParsimonyGit/frappe/commit/bac4e9b7061fd77ecb3038b2b557b132056ad91c))


### Performance Improvements

* Faster address query with explicit joins (backport #20537) (#20540) ([2ef7ef5](https://github.com/ParsimonyGit/frappe/commit/2ef7ef50519e578ce4e31c6e077439a20294deed))


### Tests

* Kanban Test fails to remove System Manager role (#20505) (#20551) ([e3e8a6b](https://github.com/ParsimonyGit/frappe/commit/e3e8a6b347d9ea2bba6c46e096695c9401181850))


### Contributors

<a href="https://github.com/mergify">mergify[bot]</a>, <a href="https://github.com/barredterra">barredterra</a>, <a href="https://github.com/sagarvora">sagarvora</a>, <a href="https://github.com/pstuhlmueller">pstuhlmueller</a>, <a href="https://github.com/surajshetty3416">surajshetty3416</a>, <a href="https://github.com/ankush">ankush</a>, <a href="https://github.com/shariquerik">shariquerik</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

ankush committed:
Merge pull request #20651 from frappe/version-14-hotfix

chore: release v14

frappe-bot committed:
chore(release): Bumped to Version 14.32.0

# [14.32.0](https://github.com/frappe/frappe/compare/v14.31.0...v14.32.0) (2023-04-11)

### Bug Fixes

* also build jsx files ([#20624](https://github.com/frappe/frappe/issues/20624)) ([#20628](https://github.com/frappe/frappe/issues/20628)) ([01f4c38](https://github.com/frappe/frappe/commit/01f4c387f3aabe5c3f77b400adf5eaf9f0d03ad5))
* better permission error for query_report ([#20643](https://github.com/frappe/frappe/issues/20643)) ([#20647](https://github.com/frappe/frappe/issues/20647)) ([33be129](https://github.com/frappe/frappe/commit/33be129e6a4dce09ae0182c3fdd008ad850b99f4))
* change z-index of freeze component to make it appear above all components ([f6ebae9](https://github.com/frappe/frappe/commit/f6ebae9117f2576617c0d8366a7017e2663823a7)), closes [#20538](https://github.com/frappe/frappe/issues/20538)
* child row form should be above freeze screen ([8d4402e](https://github.com/frappe/frappe/commit/8d4402e98bc8c7e123e42b2c14619ff85ad4b3f3))
* Consider global default in user perm ([87e1aee](https://github.com/frappe/frappe/commit/87e1aee8ca18d4d8903bc3ffde3a92b8c1d5d01b))
* Consider user perimission default in get_user_default function ([24f6931](https://github.com/frappe/frappe/commit/24f6931b0d75c5f90ee8740d9652e2817cdcdfa8))
* dont create communication for auto repeat notification ([#20606](https://github.com/frappe/frappe/issues/20606)) ([#20611](https://github.com/frappe/frappe/issues/20611)) ([e9fc2ae](https://github.com/frappe/frappe/commit/e9fc2ae8c0bc4a80c0815e15b6c8f1fd5fb46584))
* german translation of workflow state ([#20609](https://github.com/frappe/frappe/issues/20609)) ([#20616](https://github.com/frappe/frappe/issues/20616)) ([4d7a080](https://github.com/frappe/frappe/commit/4d7a080a948a728718f17235da9e15209949de36))
* **grid row:** fix prettier hook check ([583e1d4](https://github.com/frappe/frappe/commit/583e1d4062d4031a4827b02f7e7763a9df5b6743))
* **grid row:** fix update_docfield_property function not updating grid row ([a2e9cc0](https://github.com/frappe/frappe/commit/a2e9cc0240996dc3091e22cdf4cde2915bd15d68))
* improved validation in `add_comment` ([#20520](https://github.com/frappe/frappe/issues/20520)) ([4d920fa](https://github.com/frappe/frappe/commit/4d920fa380a151020e0bead2f99e340f9685908c))
* **ListView:** Evaluate sort_field sort_order within listviews based on DocTypes Definition ([#20482](https://github.com/frappe/frappe/issues/20482)) ([b560ccd](https://github.com/frappe/frappe/commit/b560ccd7bcc8ae05d2616b6afc2a88d4b74b0a1a))
* log requests even if no response ([#20638](https://github.com/frappe/frappe/issues/20638)) ([#20640](https://github.com/frappe/frappe/issues/20640)) ([a43c11e](https://github.com/frappe/frappe/commit/a43c11ebac04c6203e85d12a5f2ec098d2719e26))
* minor oauth doctypes enhancements ([#20613](https://github.com/frappe/frappe/issues/20613)) ([75300f7](https://github.com/frappe/frappe/commit/75300f7d73f7b8c6e7a64d074e4855a3ca1c18dd))
* **pretty-date:** plural form when the value is 1 ([#20619](https://github.com/frappe/frappe/issues/20619)) ([#20642](https://github.com/frappe/frappe/issues/20642)) ([7f6a557](https://github.com/frappe/frappe/commit/7f6a557d53cd4d3b31a359472870e84e914b3b57))
* track workspace changes ([#20649](https://github.com/frappe/frappe/issues/20649)) ([#20650](https://github.com/frappe/frappe/issues/20650)) ([5f54325](https://github.com/frappe/frappe/commit/5f543254f23a46b0810b6f3a7c8c8fc4ab23cbef))
* validate if doctype exists before syncing customisations (backport [#20598](https://github.com/frappe/frappe/issues/20598)) ([#20645](https://github.com/frappe/frappe/issues/20645)) ([9a39bf8](https://github.com/frappe/frappe/commit/9a39bf8daeab6fe4b83ea81a145c248cc41f363b))

### Features

* make report name translatable ([#20608](https://github.com/frappe/frappe/issues/20608)) ([#20618](https://github.com/frappe/frappe/issues/20618)) ([ed49e83](https://github.com/frappe/frappe/commit/ed49e830caa385fd9cb03fcd40bc9991faee17c1))

### Performance Improvements

* Don't re-initate sessions in realtime.py ([#20577](https://github.com/frappe/frappe/issues/20577)) ([7480b86](https://github.com/frappe/frappe/commit/7480b864fd104c64841f7887a65c3ae4ffd9e2b0))

mergify[bot] committed:
fix: consider user email if send me a copy is checked (#20627) (#20655)

* fix: consider user email if send me a copy is checked

* test(communication): test cc with include sender

(cherry picked from commit 82ec29702e61ac6f36398f2045b94cd9d78a526f)

Co-authored-by: Ritwik Puri <ritwikpuri5678@gmail.com>

shariquerik committed:
fix: update property if field exist

(cherry picked from commit 66df0e9abcc62e1a538af20a9ff5ef16dc9f3109)

shariquerik committed:
fix: update property if field exist

(cherry picked from commit 66df0e9abcc62e1a538af20a9ff5ef16dc9f3109)
(cherry picked from commit 3f10e6340f1fe70678dee6b46fdd2ac4dc2b75f7)

shariquerik committed:
Merge pull request #20660 from frappe/mergify/bp/version-14-hotfix/pr-20658

fix: update property if field exist (backport #20658)

deepeshgarg007 committed:
Merge pull request #20662 from frappe/mergify/bp/version-14/pr-20660

fix: update property if field exists (#20658)

frappe-bot committed:
chore(release): Bumped to Version 14.32.1

## [14.32.1](https://github.com/frappe/frappe/compare/v14.32.0...v14.32.1) (2023-04-12)

### Bug Fixes

* update property if field exist ([a2f637b](https://github.com/frappe/frappe/commit/a2f637b48ea74935cc8fb7aaaa055a5966602be7))

mergify[bot] committed:
fix(UX): workspace breadcrumbs based on history (#20529) (#20661)

* fix(UX): Resolve breadcrumb conflicts from history

Same report can be part of 2 workspace, in which case use breadcrumbs
from last workspace.

* fix: make sure last workspace belongs to same module at least

(cherry picked from commit c5f36a49799658655fd4c0ec64692120e2a2817b)

Co-authored-by: Ankush Menat <ankush@frappe.io>

phot0n committed:
fix: use frappe.log_error in EmailServer exception handling

(cherry picked from commit c7632d0e51b4092e0138fbb48d1ef3d26112c616)

surajshetty3416 committed:
Merge pull request #20666 from frappe/mergify/bp/version-14-hotfix/pr-20665

fix: use frappe.log_error in EmailServer exception handling (backport #20665)

mergify[bot] committed:
refactor: use urljoin to build picture url (#20664) (#20668)

(cherry picked from commit f46d1aefa9fce9f5ef04e6712a85a41f9bdfbb35)

Co-authored-by: Mohammad Hussain Nagaria <34810212+NagariaHussain@users.noreply.github.com>

mergify[bot] committed:
fix: filter with capitalized operator was getting bypassed (backport #20682) (#20684)

Co-authored-by: Shariq Ansari <sharique.rik@gmail.com>

mergify[bot] committed:
fix: group buttons are overlapping (backport #20671) (#20687)

Co-authored-by: Shariq Ansari <sharique.rik@gmail.com>

mergify[bot] committed:
fix: remove guest permission from language (#20677) (#20691)

* fix: remove guest permission from language

* fix: allow "All" to select a Language

* fix: allow "All" to read a Language

(cherry picked from commit 652202132d4e1d86fe12585541ccee65385c4402)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>

ankush committed:
test: fix print view test from lang

(cherry picked from commit 76e576c83ec4a686998a356e6cca8f91b79110c0)

mergify[bot] committed:
fix: stale `frappe.local` (#20695) (#20701)

* fix: stale `frappe.local`

Co-Authored-By: Aditya Hase <aditya@adityahase.com>

* fix: force re-init in request

To ensure that any one bad request can not completely cause recurring
loop of broken requests due to bad locals, we just force-init locals on
every request.

---------

Co-authored-by: Aditya Hase <aditya@adityahase.com>
(cherry picked from commit 1e9ede40fa89b4f4ad859cd17aa72e4b3ef94530)

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
fix: allow workflow action master rename

ankush committed:
chore: isort

mergify[bot] committed:
fix(translations)!: load translation in installed app order (backport #20648) (#20705)

* fix(translations)!: load translation in installed order

- translations are loaded in apps.txt order this doesnt make much sense.
- translations are loaded from apps which aren't even installed, again
  doesn't make sense.

Breaking but necessary change.

(cherry picked from commit 361e44de1db9aac2844b0389cd244bd3506c72af)

* perf: dont cache intermediate translation files

Just caching final files for each language is enough, duplicating
doesn't help much.

(cherry picked from commit 0b8b8294836dfc949ec552e89d571b64d909dbea)

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

shariquerik committed:
fix: prioritize modal over freeze in grid-form

(cherry picked from commit 477335eb4aee1fc3987139770feee06d254e036d)

shariquerik committed:
Merge pull request #20708 from frappe/mergify/bp/version-14-hotfix/pr-20706

fix: prioritize modal over freeze in grid-form (backport #20706)

mergify[bot] committed:
fix: remove desk access for Guest in post_model_sync (#20719) (#20720)

since we added used_oauth colum in User Email, it used to fail as it wasn't present in db in pre-model
when migrating from v13 to v14

(cherry picked from commit a232c751e5424da7c4d96e9c55dbeda5b0350574)

Co-authored-by: Ritwik Puri <ritwikpuri5678@gmail.com>

surajshetty3416 committed:
fix(export): Use fieldname if label is not set

(cherry picked from commit eced6a3dc1c07fab648a122a16ab06f9946b99b2)

surajshetty3416 committed:
fix: Patch to update Background Jobs URL

(cherry picked from commit 1053fba273b9f3544d2b82766c8471dc550bc4c0)

# Conflicts:
#	frappe/patches.txt

surajshetty3416 committed:
fix: Redirect Background Jobs to RQ Jobs

(cherry picked from commit c262eaf18d9153684e6cf3548ca461e5cd41c8ed)

# Conflicts:
#	frappe/public/js/frappe/ui/toolbar/toolbar.js

harshtandiya committed:
fix: make `owner` restricted

Co-authored-by: Mohammad Hussain Nagaria <NagariaHussain@users.noreply.github.com>
(cherry picked from commit 78829441dd395f33c489e4033a533b3075fe958e)

surajshetty3416 committed:
fix: Hide dropdown menu on click

(cherry picked from commit 52d284ff0f73b9ab781c59bfcbd3709f4da10380)

surajshetty3416 committed:
fix: Show background jobs only to permitted users

(cherry picked from commit cdb943d49ae10d55cffd900f3eb5d0effc7ccbe1)

# Conflicts:
#	frappe/public/js/frappe/ui/toolbar/toolbar.js

surajshetty3416 committed:
fix: Remove Background Jobs from dropdown

(cherry picked from commit 9938ab0f447bac7a678975e0481249102691b99f)

# Conflicts:
#	frappe/patches.txt

surajshetty3416 committed:
Merge pull request #20731 from frappe/mergify/bp/version-14-hotfix/pr-20727

fix: make `owner` restricted (backport #20727)

surajshetty3416 committed:
fix: Resolve conflicts

anandbaburajan committed:
fix: allow filter values to be saved in custom report

shariquerik committed:
Merge pull request #20733 from AnandBaburajan/custom_report_save_filters_v14

surajshetty3416 committed:
Merge pull request #20732 from frappe/mergify/bp/version-14-hotfix/pr-20730

fix(UX): Miscellaneous (backport #20730)

mergify[bot] committed:
fix: bump poplib limit (#20739) (#20740)

* fix: bump poplib limit

* Update receive.py

[skip ci]

(cherry picked from commit a2f40290d8f742795e654ad00810b6bd5fc167ae)

Co-authored-by: Ankush Menat <ankush@frappe.io>

surajshetty3416 committed:
fix: Load doc before checking permission

(cherry picked from commit ee97800f8c157170b33deacf2a5e0cc0d7ca8ee7)

# Conflicts:
#	frappe/public/js/frappe/ui/toolbar/toolbar.js

surajshetty3416 committed:
fix: Resolve conflicts

surajshetty3416 committed:
Merge pull request #20743 from frappe/mergify/bp/version-14-hotfix/pr-20742

fix: Load doc before checking permission (backport #20742)

surajshetty3416 committed:
Merge pull request #20737 from frappe/mergify/bp/version-14-hotfix/pr-20452

shariquerik committed:
fix: route to form if name has slashes in it

(cherry picked from commit c083ea3c6219084e7328b3c3587324478a387722)

shariquerik committed:
Merge pull request #20745 from frappe/mergify/bp/version-14-hotfix/pr-20744

fix: route to form if name has slashes in it (backport #20744)

mergify[bot] committed:
fix: check communication subject existence before converting to text  (#20738) (#20748)

(cherry picked from commit 13a7d12452dd918cfbacd2b856f9df81f7e67234)

Co-authored-by: Leonard Goertz <49870752+uepselon@users.noreply.github.com>

mergify[bot] committed:
chore: improve insights banner message (#20736) (#20749)

(cherry picked from commit c1af128abae09123ae1761af2f5e38c3bab3aa49)

Co-authored-by: Saqib Ansari <nextchamp.saqib@gmail.com>

mergify[bot] committed:
fix: clear all filters when in list view (backport #20342) (#20746)

mergify[bot] committed:
fix: format filter values in report print view (backport #20717) (#20752)

Co-authored-by: Michael Weißer <michael.weisser@rindenmuehle.de>
Co-authored-by: P-Froggy <60393001+P-Froggy@users.noreply.github.com>

phot0n committed:
Merge pull request #20756 from frappe/version-14-hotfix

chore: release v14

frappe-bot committed:
chore(release): Bumped to Version 14.33.0

# [14.33.0](https://github.com/frappe/frappe/compare/v14.32.1...v14.33.0) (2023-04-18)

### Bug Fixes

* allow filter values to be saved in custom report ([295d2b8](https://github.com/frappe/frappe/commit/295d2b883c9378972763361409f855a136655b6e))
* allow workflow action master rename ([2730f22](https://github.com/frappe/frappe/commit/2730f22e15c00e2c587e9774ba14427aa6a68802))
* bump poplib limit ([#20739](https://github.com/frappe/frappe/issues/20739)) ([#20740](https://github.com/frappe/frappe/issues/20740)) ([a58ef7a](https://github.com/frappe/frappe/commit/a58ef7ad46ce30becdf918672bdb6e5e25e04474))
* check communication subject existence before converting to text  ([#20738](https://github.com/frappe/frappe/issues/20738)) ([#20748](https://github.com/frappe/frappe/issues/20748)) ([3d047b4](https://github.com/frappe/frappe/commit/3d047b4cecd88c747b215e92ae38a533300ba769))
* clear all filters when in list view (backport [#20342](https://github.com/frappe/frappe/issues/20342)) ([#20746](https://github.com/frappe/frappe/issues/20746)) ([54d0dca](https://github.com/frappe/frappe/commit/54d0dcac4eb242fef2b0126e6a0f8e525b5a3752))
* consider user email if send me a copy is checked ([#20627](https://github.com/frappe/frappe/issues/20627)) ([#20655](https://github.com/frappe/frappe/issues/20655)) ([3dc215d](https://github.com/frappe/frappe/commit/3dc215d3317df802060c9edd66a6fa373bfa15ef))
* **export:** Use fieldname if label is not set ([f497557](https://github.com/frappe/frappe/commit/f497557fada3350fcff4f1ba4aead1b438a2b81c))
* filter with capitalized operator was getting bypassed (backport [#20682](https://github.com/frappe/frappe/issues/20682)) ([#20684](https://github.com/frappe/frappe/issues/20684)) ([7831866](https://github.com/frappe/frappe/commit/783186652dd8702d7e38a3f3f76d9bf80adf01ab))
* format filter values in report print view (backport [#20717](https://github.com/frappe/frappe/issues/20717)) ([#20752](https://github.com/frappe/frappe/issues/20752)) ([83fdb2e](https://github.com/frappe/frappe/commit/83fdb2e4e9af63fdeb9dc421b2a49273e29bdbd8))
* group buttons are overlapping (backport [#20671](https://github.com/frappe/frappe/issues/20671)) ([#20687](https://github.com/frappe/frappe/issues/20687)) ([149eed7](https://github.com/frappe/frappe/commit/149eed72bb3f8648d612f85f4646aa547bc8bd1f))
* Hide dropdown menu on click ([454f3a7](https://github.com/frappe/frappe/commit/454f3a7c9c626eaf6bb256f948a8c3704cbfd9c5))
* Load doc before checking permission ([a8348ce](https://github.com/frappe/frappe/commit/a8348cedd89115dcd8a81d03bedd92904bb9e1b8))
* make `owner` restricted ([8ad7ddb](https://github.com/frappe/frappe/commit/8ad7ddb58d3ce59616d6d0cd03dfd87abc7ab9d8))
* Patch to update Background Jobs URL ([67add11](https://github.com/frappe/frappe/commit/67add116bb80ae22b8e4dfebbc64126cba1e1e6b))
* prioritize modal over freeze in grid-form ([953b331](https://github.com/frappe/frappe/commit/953b331e2b4a245e0b1c65cd4d4089870d63e325))
* Redirect Background Jobs to RQ Jobs ([2cc1092](https://github.com/frappe/frappe/commit/2cc1092bee256108ba7b4284ddd064ad7adac8f6))
* Remove Background Jobs from dropdown ([0c7009c](https://github.com/frappe/frappe/commit/0c7009c0992a00b59015a0a7616a1d9e9a8a2243))
* remove desk access for Guest in post_model_sync ([#20719](https://github.com/frappe/frappe/issues/20719)) ([#20720](https://github.com/frappe/frappe/issues/20720)) ([97e0836](https://github.com/frappe/frappe/commit/97e0836ba7388f9aec9518d60c797bba19f861b7))
* remove guest permission from language ([#20677](https://github.com/frappe/frappe/issues/20677)) ([#20691](https://github.com/frappe/frappe/issues/20691)) ([de3b484](https://github.com/frappe/frappe/commit/de3b484a918aa2feb1af85eea4febdf78dad8c5a))
* Resolve conflicts ([a66fa64](https://github.com/frappe/frappe/commit/a66fa64b9df3de0b4761e843fce88e670b025bd7))
* Resolve conflicts ([abacaba](https://github.com/frappe/frappe/commit/abacaba96c6f31f464a2efd144d9c549e92f0ecf))
* route to form if name has slashes in it ([ae7d693](https://github.com/frappe/frappe/commit/ae7d69372ac88753094fc11418ad034ee77b32dc))
* Show background jobs only to permitted users ([c065b2b](https://github.com/frappe/frappe/commit/c065b2b893da810c607cb58229d6ba1b3b6e3f4c))
* stale `frappe.local` ([#20695](https://github.com/frappe/frappe/issues/20695)) ([#20701](https://github.com/frappe/frappe/issues/20701)) ([2715d2d](https://github.com/frappe/frappe/commit/2715d2d9bdddd87d67810801b4ce8effede5ff4b))
* update property if field exist ([3f10e63](https://github.com/frappe/frappe/commit/3f10e6340f1fe70678dee6b46fdd2ac4dc2b75f7))
* use frappe.log_error in EmailServer exception handling ([25281e3](https://github.com/frappe/frappe/commit/25281e3132325ee060862b736073ecceea6ee2ea))
* **UX:** workspace breadcrumbs based on history ([#20529](https://github.com/frappe/frappe/issues/20529)) ([#20661](https://github.com/frappe/frappe/issues/20661)) ([260ef71](https://github.com/frappe/frappe/commit/260ef718674d23da3183eeeb37615a2d1cea4401))

### Features

* redis cache decorator ([#20452](https://github.com/frappe/frappe/issues/20452)) ([3dff558](https://github.com/frappe/frappe/commit/3dff5588bc873a016debbdad8714217949a102d0))

shariquerik committed:
fix: charts on dashboard not loading

(cherry picked from commit 93f7fa1883fc9fc6012ef38211aeeb918105b469)
(cherry picked from commit 66ebda6380c7f64ef0e85c88a0d489776d017e07)

shariquerik committed:
Merge pull request #20796 from frappe/mergify/bp/version-14/pr-20795

fix: charts on dashboard not loading (backport #20793) (backport #20795)

frappe-bot committed:
chore(release): Bumped to Version 14.33.1

## [14.33.1](https://github.com/frappe/frappe/compare/v14.33.0...v14.33.1) (2023-04-20)

### Bug Fixes

* charts on dashboard not loading ([6103b9a](https://github.com/frappe/frappe/commit/6103b9a2088cd28863e99279dd585ab41b45fd0c))

Commit:
Merge remote-tracking branch 'frappe/version-14' into merge-v14
